ggshield/sca/output/schemas.py

```diff
@@ -0,0 +1,78 @@
+from dataclasses import dataclass, field
+from datetime import datetime
+from typing import Literal, Optional, cast
+
+import marshmallow_dataclass
+from pygitguardian.models import Base, BaseSchema, FromDictMixin
+
+
+@dataclass
+class SCAVulnerability(Base, FromDictMixin):
+    severity: str
+    summary: str
+    cve_ids: list[str] = field(default_factory=list)
+    created_at: Optional[datetime] = None
+    fixed_version: Optional[str] = None
+
+
+SCAVulnerability.SCHEMA = cast(
+    BaseSchema,
+    marshmallow_dataclass.class_schema(SCAVulnerability, base_schema=BaseSchema)(),
+)
+
+SCADependencyType = Literal["direct", "transitive"]
+
+
+@dataclass
+class SCAVulnerablePackageVersion(Base, FromDictMixin):
+    package_full_name: str
+    version: str
+    ecosystem: str
+    dependency_type: Optional[SCADependencyType] = None
+    vulns: list[SCAVulnerability] = field(default_factory=list)
+
+
+SCAVulnerablePackageVersion.SCHEMA = cast(
+    BaseSchema,
+    marshmallow_dataclass.class_schema(
+        SCAVulnerablePackageVersion, base_schema=BaseSchema
+    )(),
+)
+
+
+@dataclass
+class SCALocationVulnerability(Base, FromDictMixin):
+    location: str
+    package_vulns: list[SCAVulnerablePackageVersion] = field(default_factory=list)
+
+
+SCALocationVulnerability.SCHEMA = cast(
+    BaseSchema,
+    marshmallow_dataclass.class_schema(
+        SCALocationVulnerability, base_schema=BaseSchema
+    )(),
+)
+
+
+@dataclass
+class SCAScanAllOutput(Base, FromDictMixin):
+    scanned_files: list[str] = field(default_factory=list)
+    found_package_vulns: list[SCALocationVulnerability] = field(default_factory=list)
+
+
+SCAScanAllOutput.SCHEMA = cast(
+    BaseSchema,
+    marshmallow_dataclass.class_schema(SCAScanAllOutput, base_schema=BaseSchema)(),
+)
+
+
+class SCAScanDiffOutput(Base, FromDictMixin):
```

Comment:
You're missing a `@dataclass` here, I think

Reply:
Thanks, done